src/lib/hooks/useFrames.svelte.ts

```diff
@@ -121,67 +106,52 @@ export const provideFrames = (partID: () => string) => {
 			)
 
 			if (setComponentModIndex < unsetComponentModIndex) {
-				unsetObjects.push(fragmentComponentName)
+				unsetResults.push(fragmentComponentName)
 			} else if (unsetComponentModIndex < setComponentModIndex) {
 				const frameData = fragmentMod.mods[setComponentModIndex]['$set'][
 					`components.${fragmentComponentName}.frame`
 				] as Frame
-				const pose = createPoseFromFrame(frameData)
-				const geometry = createGeometryFromFrame(frameData)
-				const worldObject = new WorldObject(fragmentComponentName, pose, frameData.parent, geometry)
-				objects.push(worldObject)
+				results[fragmentComponentName] = createTransformFromFrame(fragmentComponentName, frameData)
 			}
 		}
-		return [objects, unsetObjects]
+		return [results, unsetResults]
 	})
 
-	$effect.pre(() => {
-		for (const frame of configFrames) {
-			const result = machineFrames[frame.name]
-
-			if (result) {
-				result.referenceFrame = frame.referenceFrame
-				result.localEditedPose = frame.pose
-				result.geometry = frame.geometry
-			} else {
-				machineFrames[frame.name] = frame
-			}
+	const frames = $derived.by(() => {
```

Comment:
Another benefit of this approach is that deduping is simply done by merging objects. Also all $effects have been replaced with $deriveds